disperser/batcher/encoding_streamer.go

```diff
@@ -510,7 +511,8 @@ func (e *EncodingStreamer) CreateBatch() (*batch, error) {
 		i++
 	}
 
-	state, err := e.getOperatorStateForBlobs(context.Background(), metadatas, e.ReferenceBlockNumber)
+	// metadatas here should have been already validated
+	state, _, err := e.getOperatorStateForValidBlobs(context.Background(), metadatas, e.ReferenceBlockNumber)
```

Comment:
I guess you're happy with rerunning the unnecessary validation here?

Reply:
I had reused the method for simplicity, but I'll just separate out this method into two: one for getting the state and the other for validation/filtering metadata